DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```
@@ -134,48 +130,162 @@ function ValidateProperties
                 $Parameters["InterfaceAlias"] = $currentIP[0].InterfaceAlias
+                if($DefaultGateway)
+                {
+                    $Parameters["DefaultGateway"] = $DefaultGateway
+                }
 
-                if($DefaultGateway){ $Parameters["DefaultGateWay"] = $DefaultGateway }
+                # If null, remove the gateway from the current settings hash table
+                if($currentSettings['DefaultGateway'] -eq $null)
+                {
+                    $currentSettings.remove('DefaultGateway')
+                }
+
+                # Remove IP address first; required if the IP is correct but other settings are not
+                Remove-NetIPAddress @currentSettings -confirm:$false -ErrorAction Stop
+
+                # Apply the specified IP configuration
                 $null = New-NetIPAddress @Parameters -ErrorAction Stop
 
                 # Make the connection profile private
-                Get-NetConnectionProfile -InterfaceAlias $InterfaceAlias | Set-NetConnectionProfile -NetworkCategory Private -ErrorAction SilentlyContinue
-                Write-Verbose -Message "IPAddress is set to $IPAddress."
+                Get-NetConnectionProfile -InterfaceAlias $InterfaceAlias | `
+                    Set-NetConnectionProfile -NetworkCategory Private -ErrorAction SilentlyContinue
+                Write-Verbose -Message "IP Interface was set to the desired state."
+            }
+            else
+            {
+                return $false
             }
-            else {return $false}
         }
         else
         {
-            Write-Verbose -Message "IPAddress is correct."
+            Write-Verbose -Message "IP Interface is in the desired state."
             return $true
         }
     }
     catch
     {
-       Write-Verbose -Message $_
-       throw "Can not set or find valid IPAddress using InterfaceAlias $InterfaceAlias and AddressFamily $AddressFamily"
+        Write-Verbose -Message $_
```

I would caution against throwing your own exception here. If we've run into a problem you are now masking it instead of letting PowerShell throw a more obvious exception.  Honestly I'd probably remove the try/catch and let it fail in its own right considering we'll stop at the failure point if its a bad enough of an exception.

Or at least rethrow whatever the current exception is instead of writing your own.